erts/emulator/beam/jit/beam_jit_common.c

```diff
@@ -0,0 +1,756 @@
+                                                    reg,
+                                                    bif_nif_arity);
+    }
+
+    return nif_bif_result;
+}
+
+enum beam_jit_nif_load_ret beam_jit_load_nif(Process *c_p,
+                                             ErtsCodePtr I,
+                                             Eterm *reg) {
+    if (erts_try_seize_code_write_permission(c_p)) {
+        Eterm result;
+
+        PROCESS_MAIN_CHK_LOCKS((c_p));
+        ERTS_UNREQ_PROC_MAIN_LOCK((c_p));
+        result = erts_load_nif(c_p, I, reg[0], reg[1]);
+        erts_release_code_write_permission();
+        ERTS_REQ_PROC_MAIN_LOCK(c_p);
+
+        if (ERTS_LIKELY(is_value(result))) {
+            reg[0] = result;
+            return RET_NIF_success;
+        } else {
+            c_p->freason = BADARG;
+            return RET_NIF_error;
+        }
+    } else {
+        /* Yield and try again. */
+        c_p->current = NULL;
+        c_p->arity = 2;
+        return RET_NIF_yield;
+    }
+}
+
+Uint beam_jit_get_map_elements(Eterm map,
+                               Eterm *reg,
+                               Eterm *E,
+                               Uint n,
+                               Eterm *fs) {
+    Uint sz;
+
+    /* This instruction assumes Arg1 is a map, i.e. that it follows a test
+     * is_map if needed. */
+
+    if (is_flatmap(map)) {
+        flatmap_t *mp;
+        Eterm *ks;
+        Eterm *vs;
+
+        mp = (flatmap_t *)flatmap_val(map);
+        sz = flatmap_get_size(mp);
+
+        if (sz == 0) {
+            return 0;
+        }
+
+        ks = flatmap_get_keys(mp);
+        vs = flatmap_get_values(mp);
+
+        while (sz) {
+            if (EQ(fs[0], *ks)) {
+                PUT_TERM_REG(*vs, fs[1]);
+
+                n--;
+                fs += 3;
+
+                /* no more values to fetch, we are done */
+                if (n == 0) {
+                    return 1;
+                }
+            }
+
+            ks++, sz--, vs++;
+        }
+        return 0;
+    } else {
+        ASSERT(is_hashmap(map));
+
+        while (n--) {
+            const Eterm *v;
+            Uint32 hx;
+
+            hx = fs[2];
+            ASSERT(hx == hashmap_make_hash(fs[0]));
+
+            if ((v = erts_hashmap_get(hx, fs[0], map)) == NULL) {
+                return 0;
+            }
+
+            PUT_TERM_REG(*v, fs[1]);
+            fs += 3;
+        }
+
+        return 1;
+    }
+}
+
+static void TEST_BIN_VHEAP(Process *c_p,
+                           Eterm *reg,
+                           Uint VNh,
+                           Uint Nh,
+                           Uint Live) {
+    int need = Nh;
+
+    if (c_p->stop - c_p->htop < (need + S_RESERVED) ||
+        MSO(c_p).overhead + VNh >= BIN_VHEAP_SZ(c_p)) {
+        c_p->fcalls -=
+                erts_garbage_collect_nobump(c_p, need, reg, Live, c_p->fcalls);
+    }
+}
+
+static void GC_TEST(Process *c_p, Eterm *reg, Uint Ns, Uint Nh, Uint Live) {
```

Comment:
Maybe rename these to `beam_jit_gc_test` and `beam_jit_test_bin_vheap`?

Reply:
They don't need `beam_jit` prefix because they are `static`, but I will spell their names in lowercase. I assume that the only reason they were spelled in uppercase in the first place is because they inherited their names from macros.